modules/jfx.incubator.richtext/src/main/java/jfx/incubator/scene/control/richtext/TextPos.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2022, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jfx.incubator.scene.control.richtext;
+
+/**
+ * An immutable position within the text.
+ * <p>
+ * Because it is immutable, it cannot track locations in the document which is being edited.
+ * For that, use {@link Marker}.
+ */
+public final class TextPos implements Comparable<TextPos> {
+    /** A text position at the start of the document. */
+    public static final TextPos ZERO = new TextPos(0, 0, 0, true);
+    private final int index;
+    private final int offset;
+    private final int charIndex;
+    private final boolean leading;
+
+    /**
+     * Creates a new text position.
+     * @param index the paragraph index
+     * @param offset the text offset
+     * @param charIndex the character index
+     * @param leading true if leading
```

Comment:
This needs a better description of what the parameters mean. I can't tell from just reading the docs what the difference is between "offset" and "charIndex". And you don't define "leading" -- I presume you mean related to the cursor? If so, what is its meaning if the TextPos isn't being used as the position of a cursor?

Reply:
Please see if the new description is better.  This class is similar to HitInfo.